### PR TITLE
fix: resizing in freescroll

### LINKED ISSUE #177: When resize is called in "freeScroll" move type, panel is aligned.
## Description
3e2cc1cc-6084-11e9-9dfe-cc73e738b0c4
> `autoResize` is enabled

## Steps to check or reproduce
1. Init Flicking instance in "freeScroll" move type.
2. Call resize() when panel is not aligned.

### LINKED ISSUE #178: Changing DOM element's size in FlickingPanel's update() method breaks layout.
## Description
In FlickingPanel's `update()` method, any callback that changes panel's size is not applied automatically and breaks layout.

## Steps to check or reproduce
```js
panel.update(el => {
  // Do anything that changes panel's size, like changing CSS, innerHTML...
});

### PR DESCRIPTION
**Pull request by WoodNeck**
## Issue
This fixes #177 
This also fixes #178 

## Details
This PR fixes both #177 , #178
Also it adds some missing tests.

### CODE REVIEW

File: src/components/Viewport.ts [hunk 1]
```diff
@@ -844,7 +850,7 @@ export default class Viewport {
     // Update panel position && fit to wrapper
     let nextPanelPos = firstPanel.getPosition();
     let maintainingPanel: Panel = firstPanel;
-    if ((currentState.holding || currentState.playing) && nearestPanel) {
+    if (nearestPanel) {
```
Comment (daybrush) [inline src/components/Viewport.ts]: @WoodNeck 

What kind of conditions is this?

File: test/unit/methods.spec.ts [hunk 2]
```diff
@@ -740,6 +741,28 @@ describe("Methods call", () => {
...
+      const prevCameraPosition = viewport.getCameraPosition();
+
+      // When
+      flicking.resize();
+
+      // Then
+      const afterCameraPosition = viewport.getCameraPosition();
+      expect(prevCameraPosition).equals(afterCameraPosition);
+    });
   });
```
Comment (daybrush) [inline test/unit/methods.spec.ts]: @WoodNeck 
Wouldn't it be nice to have a test when the pos goes out of range?

File: src/components/Viewport.ts [hunk 3]
```diff
@@ -395,6 +396,11 @@ export default class Viewport {
 
     this.resize();
 
+    const isFreeScroll = (this.options.moveType as MoveTypeObjectOption).type === "freeScroll";
+    if (isFreeScroll && wasEmpty) {
+      this.moveTo(this.currentPanel!, "", null, 0);
```
Comment (jongmoon) [inline src/components/Viewport.ts]: @WoodNeck Is above code related with `resizing`?

File: src/components/Viewport.ts (hunk 1 above)
Comment (WoodNeck) [reply src/components/Viewport.ts]: This was intended to update panel positions correctly only when in holding/animating state.
But in free scroll mode, it was also needed that it should be updated same in the idle state.

File: src/components/Viewport.ts (hunk 3 above)
Comment (WoodNeck) [reply src/components/Viewport.ts]: Nope, it is related with adding new panel when there were no panels in free scroll mode.
As free scroll maintains camera position while in resizing, camera position usually is wrong after adding new panel.
So, I've manually modified it for its circumstance.

File: test/unit/methods.spec.ts (hunk 2 above)
Comment (WoodNeck) [reply test/unit/methods.spec.ts]: That can be temporary fix.
To fix it correctly, we should modify Axes first so we can animating position as our own.
I think we should be able to change camera position while resizing first.

**Pull request merged**